android/src/main/java/com/reactnativeescposprinter/EscPosPrinterDiscoveryModule.java

```diff
@@ -35,211 +37,220 @@
-              // location requests here.
-              // ...
-              promise.resolve("SUCCESS");
-          }
-      });
-
-      task.addOnFailureListener(new OnFailureListener() {
-          @Override
-          public void onFailure(@NonNull Exception e) {
-              if (e instanceof ResolvableApiException) {
+    task.addOnFailureListener(new OnFailureListener() {
+      @Override
+      public void onFailure(@NonNull Exception e) {
+        if (e instanceof ResolvableApiException) {
 
-                  // Location settings are not satisfied, but this can be fixed
-                  // by showing the user a dialog.
-                  try {
-                      // Show the dialog by calling startResolutionForResult(),
-                      // and check the result in onActivityResult().
+          // Location settings are not satisfied, but this can be fixed
+          // by showing the user a dialog.
+          try {
+            // Show the dialog by calling startResolutionForResult(),
+            // and check the result in onActivityResult().
 
-                      ResolvableApiException resolvable = (ResolvableApiException) e;
-                      resolvable.startResolutionForResult(currentActivity,
-                              CommonStatusCodes.RESOLUTION_REQUIRED);
+            ResolvableApiException resolvable = (ResolvableApiException) e;
+            resolvable.startResolutionForResult(currentActivity, CommonStatusCodes.RESOLUTION_REQUIRED);
 
-                  } catch (IntentSender.SendIntentException sendEx) {
-                    promise.reject("ERROR");
-                  }
-              }
+          } catch (IntentSender.SendIntentException sendEx) {
+            promise.reject("ERROR");
           }
-      });
+        }
+      }
+    });
   }
 
-  private void sendEvent(ReactApplicationContext reactContext,
-    String eventName,
-    @Nullable WritableArray params) {
-      reactContext
-      .getJSModule(DeviceEventManagerModule.RCTDeviceEventEmitter.class)
-      .emit(eventName, params);
+  private void sendEvent(ReactApplicationContext reactContext, String eventName, @Nullable WritableArray params) {
+    reactContext.getJSModule(DeviceEventManagerModule.RCTDeviceEventEmitter.class).emit(eventName, params);
   }
 
   private void stopDiscovery() {
     while (true) {
       try {
         Discovery.stop();
         break;
-      }
-      catch (Epos2Exception e) {
+      } catch (Epos2Exception e) {
         if (e.getErrorStatus() != Epos2Exception.ERR_PROCESSING) {
           return;
         }
       }
     }
   }
 
-  private void startDiscovery(MyCallbackInterface callback) {
-     this.stopDiscovery();
+  private void startDiscovery(MyCallbackInterface callback, ReadableMap paramsMap) {
+    this.stopDiscovery();
 
-     FilterOption mFilterOption = new FilterOption();
+    if (paramsMap != null) {
+      mExtractSerialNumber = paramsMap.getBoolean("serialNumber");
```

Comment:
Yes, that would be a better name for the option.
The option is already optional at the RN/typescript side.

Reply:
Yeah but also It would be nice to make all params object optional. I mean
```
discover() // no errors
discover({}) // no errors
discover({unsupprotedoption: true }) // TS error
discover({usbSerialNumber: true }) // no errors
```

Reply:
The latest commit fixes crash in Java when using `discover({})`

Reply:
Ok it's expected